pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -196,11 +196,11 @@ func _ipSecReplacePolicyInFwd(src, dst *net.IPNet, tmplSrc, tmplDst net.IP, prox
 	policy := ipSecNewPolicy()
 	policy.Dir = dir
 	policy.Dst = dst
-	policy.Mark = &netlink.XfrmMark{
-		Mask: linux_defaults.IPsecMarkMaskIn,
-	}
 	if dir == netlink.XFRM_DIR_IN {
 		policy.Src = src
+		policy.Mark = &netlink.XfrmMark{
```

Comment:
I surely don't have the whole knowledge, based on your well descriptive commit message my understanding is that we should not mark packet for `dir == netlink.XFRM_DIR_FWD`. You decided to move the mark under `dir == netlink.XFRM_DIR_IN` which is right (for sure :-)), but should not be more specific with condition like:

```
if dir != netlink.XFRM_DIR_FWD` {
  // We want to mark packet...
  policy.Mark = ...
}
```

Based on the current algorithm I'm worry to see `dir` with something else than `XFRM_DIR_IN` and basically have packet not marked.

Reply:
I don't think it's generally a good idea to make the code more convoluted to better match what the commit description says.

More specifically, in this case, (1) I would be very surprised if we need to support a new direction besides In, Out, and Fwd (it's baked in the kernel and netfilter) and (2) if that ever happened, I would argue that not matching on the mark would likely be a better default.

Reply:
Fair enough, thank you Paul. I would have suggested to at least add a comment  but I imagine we could use git blame to understand the reason of this move. ++